src/email/plotting.py

```diff
@@ -0,0 +1,527 @@
+import pytz
+from matplotlib import pyplot as plt
+
+from src.constants import (
+    CERF_SIDS,
+    CHD_GREEN,
+    D_THRESH,
+    SPANISH_MONTHS,
+    LON_ZOOM_RANGE,
+    PROJECT_PREFIX,
+    THRESHS,
+)
+from src.datasources import codab, nhc
+from src.email.utils import (
+    TEST_FCAST_MONITOR_ID,
+    TEST_OBSV_MONITOR_ID,
+    TEST_STORM,
+    add_test_row_to_monitoring,
+    open_static_image,
+)
+from src.monitoring import monitoring_utils
+import ocha_stratus as stratus
+
+
+def get_plot_blob_name(monitor_id, plot_type: Literal["map", "scatter"]):
+    fcast_obsv = "fcast" if "fcast" in monitor_id.lower() else "obsv"
+    return (
+        f"{PROJECT_PREFIX}/plots/{fcast_obsv}/" f"{monitor_id}_{plot_type}.png"
+    )
+
+
+def convert_datetime_to_es_str(x: pd.Timestamp) -> str:
+    es_str = x.strftime("%Hh%M, %-d %b")
+    for en_mo, es_mo in SPANISH_MONTHS.items():
+        es_str = es_str.replace(en_mo, es_mo)
+    return es_str
+
+
+def update_plots(
+    fcast_obsv: Literal["fcast", "obsv"],
+    clobber: list = None,
+    verbose: bool = False,
+):
+    if clobber is None:
+        clobber = []
+    df_monitoring = monitoring_utils.load_existing_monitoring_points(
+        fcast_obsv
+    )
+    if TEST_STORM:
+        df_monitoring = add_test_row_to_monitoring(df_monitoring, fcast_obsv)
+    existing_plot_blobs = stratus.list_container_blobs(
+        name_starts_with=f"{PROJECT_PREFIX}/plots/{fcast_obsv}/"
+    )
+
+    for monitor_id, row in df_monitoring.set_index("monitor_id").iterrows():
+        for plot_type in ["map", "scatter"]:
+            blob_name = get_plot_blob_name(monitor_id, plot_type)
+            if blob_name in existing_plot_blobs and plot_type not in clobber:
+                if verbose:
+                    print(f"Skipping {blob_name}, already exists")
+                continue
+            print(f"Creating {blob_name}")
+            create_plot(monitor_id, plot_type, fcast_obsv)
+
+
+def create_plot(
+    monitor_id: str,
+    plot_type: Literal["map", "scatter"],
+    fcast_obsv: Literal["fcast", "obsv"],
+):
+    if plot_type == "map":
+        create_map_plot(monitor_id, fcast_obsv)
+    elif plot_type == "scatter":
+        create_scatter_plot(monitor_id, fcast_obsv)
+    else:
+        raise ValueError(f"Unknown plot type: {plot_type}")
+
+
+def create_scatter_plot(monitor_id: str, fcast_obsv: Literal["fcast", "obsv"]):
+    df_monitoring = monitoring_utils.load_existing_monitoring_points(
+        fcast_obsv
+    )
+    if monitor_id in [TEST_FCAST_MONITOR_ID, TEST_OBSV_MONITOR_ID]:
+        df_monitoring = add_test_row_to_monitoring(df_monitoring, fcast_obsv)
+    monitoring_point = df_monitoring.set_index("monitor_id").loc[monitor_id]
+    cuba_tz = pytz.timezone("America/Havana")
+    cyclone_name = monitoring_point["name"]
+    issue_time = monitoring_point["issue_time"]
+    issue_time_cuba = issue_time.astimezone(cuba_tz)
+    blob_name = f"{PROJECT_PREFIX}/processed/stats_{D_THRESH}km.csv"
```

Comment:
@t-downing  - can you add this file to the blob ? I think it should come from the historical analysis?